import_obfuscation.go

```diff
@@ -0,0 +1,572 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/Binject/debug/goobj2"
+)
+
+// pkgInfo stores a parsed go archive/object file,
+// and the original path to which it was read from.
+type pkgInfo struct {
+	pkg  *goobj2.Package
+	path string
+}
+
+// dataType signifies whether the Data portion of a
+// goobj2.Sym is reflection data for an import path,
+// reflection data for a method of struct field, or
+// something else.
+type dataType uint8
+
+const (
+	other dataType = iota
+	importPath
+	namedata
+)
+
+// privateImports stores package paths and names that
+// match GOPRIVATE. privateNames are elements of the
+// paths in privatePaths, separated so that the shorter
+// names don't accidently match another import, such
+// as a stdlib package
+type privateImports struct {
+	privatePaths []string
+	privateNames []string
+}
+
+func obfuscateImports(objPath, importCfgPath string) (map[string]string, error) {
+	importCfg, err := goobj2.ParseImportCfg(importCfgPath)
+	if err != nil {
+		return nil, err
+	}
+	mainPkg, err := goobj2.Parse(objPath, "main", importCfg)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing main objfile: %v", err)
+	}
+	privatePkgs := []pkgInfo{{mainPkg, objPath}}
+
+	// build list of imported packages that are private
+	for pkgPath, info := range importCfg {
+		if isPrivate(pkgPath) {
```

Comment:
`if !isPrivate(pkgPath) { continue }`

Reply:
I still have to write non garbled packages, or the linker wouldn't know where to find them